examples/Example15/electrons.cu

```diff
@@ -0,0 +1,486 @@
+    G4HepEmElectronTrack elTrack;
+    G4HepEmTrack *theTrack = elTrack.GetTrack();
+    theTrack->SetEKin(currentTrack.energy);
+    theTrack->SetMCIndex(theMCIndex);
+    theTrack->SetOnBoundary(currentTrack.navState.IsOnBoundary());
+    theTrack->SetCharge(Charge);
+    G4HepEmMSCTrackData *mscData = elTrack.GetMSCTrackData();
+    mscData->fIsFirstStep        = currentTrack.initialRange < 0;
+    mscData->fInitialRange       = currentTrack.initialRange;
+
+    // Initial value of magnetic field
+    // Equation_t::EvaluateDerivativesReturnB(magField, currentTrack.pos,
+    //                                       , momentum* currentTrack.dir,
+    //                                       , charge, dy_ds, magFieldStart);
+    
+    // Prepare a branched RNG state while threads are synchronized. Even if not
+    // used, this provides a fresh round of random numbers and reduces thread
+    // divergence because the RNG state doesn't need to be advanced later.
+    RanluxppDouble newRNG(currentTrack.rngState.BranchNoAdvance());
+
+    // Compute safety, needed for MSC step limit.
+    vecgeom::Precision safety = 0.0;
+    if (!currentTrack.navState.IsOnBoundary()) {
+      safety = BVHNavigator::ComputeSafety(currentTrack.pos, currentTrack.navState);
+    }
+    theTrack->SetSafety(safety);
+
+    RanluxppDoubleEngine rnge(&currentTrack.rngState);
+
+    // Sample the `number-of-interaction-left` and put it into the track.
+    for (int ip = 0; ip < 3; ++ip) {
+      double numIALeft = currentTrack.numIALeft[ip];
+      if (numIALeft <= 0) {
+        numIALeft                  = -std::log(currentTrack.Uniform());
+        currentTrack.numIALeft[ip] = numIALeft;
+      }
+      theTrack->SetNumIALeft(numIALeft, ip);
+    }
+
+    // Call G4HepEm to compute the physics step limit.
+    G4HepEmElectronManager::HowFar(&g4HepEmData, &g4HepEmPars, &elTrack, &rnge);
+
+    // Remember value of initial range for the next step.
+    currentTrack.initialRange = mscData->fInitialRange;
+
+    // Get result into variables.
+    double stepLengthFromPhysics = theTrack->GetGStepLength();
+    // The phyiscal step length is the amount that the particle experiences
+    // which might be longer than the geometrical step length due to MSC. As
+    // long as we call PerformContinuous in the same kernel we don't need to
+    // care, but we need to make this available when splitting the operations.
+    // double physicalStepLength = elTrack.GetPStepLength();
+    int winnerProcessIndex = theTrack->GetWinnerProcessIndex();
+    // Leave the range and MFP inside the G4HepEmTrack. If we split kernels, we
+    // also need to carry them over!
+
+    // Check if there's a volume boundary in between.
+    bool propagated = true;
+    double geometryStepLength;
+    vecgeom::NavStateIndex nextState;
+    
+    float BzFieldValue= *gPtrBzFieldValue_dev;   // Use vecgeom::Precision ?
+    if (BzFieldValue != 0.0) {
+      stepLengthFromPhysics= 0.1;
```

Comment:
I suppose this is from debugging?

Reply:
Yes, of course.  I am making revisions - I had hoped to finish and push them already yesterday, but got sidelined.